### PR TITLE
Add a storybook for the ui package

### PR DESCRIPTION
**Pull request by tmeasday**
Issue:

No storybook exists for storybook's UI itself 😮 

## What I did

Added stories for the `lib/ui` package

## How to test

Run `yarn storybook` from the top level.

I'm not sure if this is the best approach. We could also add a storybook inside `lib/ui` but we'd have a circular dependency issue.

Note as well that I added the stories for the `lib/components` package to this storybook; I note they've also been added to the `cra-kitchen-sink` app -- perhaps this is a better approach?

Let me know what you think.

### Is this testable with jest or storyshots?

Potentially we could use storyshots? 

### Does this need a new example in the kitchen sink apps?

NO

### Does this need an update to the documentation?

NO

### CODE REVIEW

File: .storybook/addons.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions */
```
Comment (ndelangen) [inline .storybook/addons.js]: Can we not use a `/examples/<any>` here? Do we really have to add something to the root of the project?

**Review by ndelangen — CHANGES_REQUESTED** [review]

File: .storybook/addons.js (hunk 1 above)
Comment (ndelangen) [reply .storybook/addons.js]: I would be happy with an `examples/official` or something, where we have a storybook to demo storybook's UI components.

**Comment by ndelangen** [comment]
Lot's of love here, This will be fantastic to have!

Just hoping we can move the files away from the project root, since we're already crowding the project root folder a lot.

**Comment by Hypnosphi** [comment]
> Note as well that I added the stories for the lib/components package to this storybook; I note they've also been added to the cra-kitchen-sink app -- perhaps this is a better approach?

In fact, I originally put them to `lib/components`, but @ndelangen asked me to move them to `cra-kitchen-sink`.
And now I actually feel very convenient to have all the examples in a single storybook instance. So maybe we should move the `lib/ui` stories there as well

File: .storybook/addons.js (hunk 1 above)
Comment (tmeasday) [reply .storybook/addons.js]: I don't have any problem with that. The only thing is potentially wanting to combine all react-based storybooks in a single storybook (components, ui, cra-vanilla).

[This works better for chromatic for instance. I'm not sure it helps much else -- maybe storyshots?]

Maybe `examples/offical-storybook` or something?

Comment (tmeasday) [reply .storybook/addons.js]: Or `examples/meta[-storybook]`?

**Comment by tmeasday** [comment]
I actually kind of feel like a `storybook/` directory at the top level would be kind of a cool thing to have?

File: .storybook/addons.js (hunk 1 above)
Comment (ndelangen) [reply .storybook/addons.js]: `examples/offical-storybook` would get my vote

**Comment by ndelangen** [comment]
I don't that will convey meaning to anyone just browsing through the code, where-as a folder `/examples/official-storybook` would.

To me, it's just another example of storybook that we use internally to develop and test our work on.

**Comment by Hypnosphi** [comment]
> I actually kind of feel like a storybook/ directory at the top level would be kind of a cool thing to have?

Maybe, but only if we move all the other sample stories there

**Comment by tmeasday** [comment]
I'm not very opinionated about this, but I would say as a new developer I would probably not expect to find the stories about the storybook UI in the "kitchen sink" example. 

I think we all agree that all the react-based stories (ui, components, various addons, other stuff in KS) should be in the one storybook. My vote would be `examples/official-storybook`. Then we can make KS simple again (like the other KSs) and move any tricky stuff like netlify over to `official-storybook`.

**Comment by Hypnosphi** [comment]
Sounds good

**Comment by ndelangen** [comment]
Let's do it!

**Comment by tmeasday** [comment]
On the case.

**Comment by tmeasday** [comment]
@ndelangen @Hypnosphi I think I did the bulk of this; a few loose ends to tie up, but let me know if this is in line with what you were thinking; I'll sort out storyshots and clean up the old cra-kitchen-sink if you are good with what i have so far.

PS the Chromatic build is starting to fill out a bit ;) https://www.chromaticqa.com/build?appId=5a375b97f4b14f0020b0cda3&number=7

**Comment by ndelangen** [comment]
That chromatic is fantastic @tmeasday !

Will Chromatic also show up as a github status?

**Comment by tmeasday** [comment]
@ndelangen sure does! Check out the chromatic PR: https://github.com/storybooks/storybook/pull/2505

screenshot 2017-12-21 11 04 53

I'll take the lack of negative comments as a positive and I'll get the tests passing on this branch.

**Comment by tmeasday** [comment]
I'm getting this error from storyshots:

```
[187:0x36cbe80]   105747 ms: Mark-sweep 1400.4 (1454.0) -> 1400.4 (1454.0) MB, 1403.1 / 0.0 ms  allocation failure GC in old space requested
[187:0x36cbe80]   107011 ms: Mark-sweep 1400.4 (1454.0) -> 1400.4 (1452.5) MB, 1262.8 / 0.0 ms  last resort GC in old space requested
[187:0x36cbe80]   108309 ms: Mark-sweep 1400.4 (1452.5) -> 1400.4 (1452.5) MB, 1297.2 / 0.0 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x3719efba5ee1 <JSObject>
    1: read [/tmp/storybook/node_modules/react-dom/cjs/react-dom-server.node.development.js:~2206] [pc=0x23f6ad3e54b2](this=0x2bc5637e721 <ReactDOMServerRenderer map = 0x51a975c0841>,bytes=0x3795986829a9 <Number inf>)
    3: renderToStaticMarkup [/tmp/storybook/node_modules/react-dom/cjs/react-dom-server.node.development.js:2512] [bytecode=0x3b8b963a9489 offset=31](this=0x1665b6d8da41 <Object ma...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x121a7ac [node]
 3: v8::Utils::ReportOOMFailure(char const*, bool) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationSpace) [node]
 6: v8::internal::Runtime_AllocateInTargetSpace(int, v8::internal::Object**, v8::internal::Isolate*) [node]
 7: 0x23f6ace0463d
```

Have you guys seen this before?

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
Definitely the right the direction @tmeasday, fantastic work!

**Comment by tmeasday** [comment]
The segfault was something to do with `react-modal`. Should probably figure it out but I'll just disable that test for storyshots for now.

**Comment by tmeasday** [comment]
@ndelangen shall I merge this?

**Comment by ndelangen** [comment]
Yes merge when ready, after, we should add a netlify site for the newly created example!

**Comment by Hypnosphi** [comment]
Why UI and components stories are not part of "official"?

**Comment by tmeasday** [comment]
How do you mean? Aren’t they?

**Comment by Hypnosphi** [comment]
Oh, now I see that they are. But their sources are in packages. We probably should use a single approach then. That is, either move `ui` and `components` stories sources to `examples/official-storybook` directory, or move all the addons' stories to corresponding packages.

CC: @ndelangen

**Comment by ndelangen** [comment]
Move all stories into the examples please.

Our packages shouldn't contain stories I think. Please let me know if you disagree on that.

I think co-locating the stories makes a bit more sense in this case. Reason for this is that we want to use the `example/*` as a way of showing off how to write stories.

**Comment by tmeasday** [comment]
Hmm, I can see the argument for this but it'd be a bit weird -- in particular the `lib/ui` stories mirror the `.test.js` files in that project (in the future it would be good to consolidate those further). When someone works on that project, wouldn't it be annoying to have to update a story file in a completely different location?

Maybe we should use a symlink or something?

**Comment by ndelangen** [comment]
🤔

**Comment by tmeasday** [comment]
I'm stumped too ;)

**Comment by tmeasday** [comment]
@ndelangen still willing to make whichever changes make sense here, I'm just not sure what's best

**Pull request merged**